service/catalog.go

```diff
@@ -53,6 +53,23 @@ type CatalogRequest struct {
 	Kind   string
 }
 
+func isDuplicateName(catalogModel *model.CatalogModel) bool {
+
+	catalogs := []model.CatalogModel{}
```

Comment:
You don't need to change this because precedence has already been set in this codebase, but in the future I'd prefer to see all calls interacting with a database pulled into their own data package and the details of interacting with the DB hidden from the consumer such that this business logic (this package and file) has no knowledge of the underlying persistence technology.